lib/Conversion/TorchToLinalg/Pooling.cpp

```diff
@@ -137,57 +162,159 @@ static LogicalResult createPoolingOp(
                              ValueRange{paddedInput, windowTensor},
                              outTensorInitialized, stridesAttr, dilationAttr)
                .getResult(0);
-
   return success();
 }
 
-
 namespace {
-class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
+template <typename OpTy, int Dim>
+class ConvertAtenMaxPoolOp : public OpConversionPattern<OpTy> {
+  using OpConversionPattern<OpTy>::OpConversionPattern;
+
+private:
+  LogicalResult writePoolingMax3D(AtenMaxPool3dOp &op,
+                                  typename OpTy::Adaptor &adaptor,
+                                  ConversionPatternRewriter &rewriter,
+                                  TypedAttr &smallestFPValueAttr, Value &self,
```

Comment:
nit: `smallestFPValueAttr` does not need to be a parameter. This function only applies to `AtenMaxPool3dOp`, so the smallest FP value should always be the same. I would just generate it inside this helper.

Reply:
Good idea! Thanks!